how-to-ns/src/com/gfredericks/how_to_ns/main.clj

```diff
@@ -54,44 +54,55 @@
         (println "Exception in how-to-ns when checking" (str file))
         (prn e)))))
 
+(def default-paths
+  ["src" "test"])
+
 (defn check
-  [paths opts]
-  (->> (all-clojure-files paths)
-       (map (fn [file]
-              (try
-                (let [contents (slurp file)
-                      formatted (how-to-ns/format-initial-ns-str contents opts)]
-                  (if (= contents formatted)
-                    0
-                    (do
-                      (binding [*out* *err*]
-                        (println "Bad ns format:")
-                        (println (unified-diff
-                                  (str file)
-                                  contents
-                                  formatted)))
-                      1)))
-                (catch Exception e
-                  (report-file-specific-exception file e)
-                  1))))
-       (reduce +)))
+  ([options]
+   (let [paths (or (:paths options) default-paths)
+         opts (merge how-to-ns/default-opts options)]
+     (check paths opts)))
+  ([paths opts]
+   (->> (all-clojure-files paths)
+        (map (fn [file]
+               (try
+                 (let [contents (slurp file)
+                       formatted (how-to-ns/format-initial-ns-str contents opts)]
+                   (if (= contents formatted)
+                     0
+                     (do
+                       (binding [*out* *err*]
+                         (println "Bad ns format:")
+                         (println (unified-diff
+                                    (str file)
+                                    contents
+                                    formatted)))
+                       1)))
+                 (catch Exception e
+                   (report-file-specific-exception file e)
+                   1))))
+        (reduce +))))
```

Comment:
What do you think about add `System/exit` in the end of `->>` thread macro? Usually i expect non-zero exit code when check failed. Then I can use it in scripts or CI.

Reply:
I agree that having that feature in general is useful and appropriate. I'm less sure about adding it here after this function has already existed for a long time, since that changes the behavior for existing users. As you can see from the bottom I imagined having a `-main` function eventually; I'm not too familiar with the conventions around when a function should just be a function vs when it should assume it's being called as a tool and can exist the whole jvm. Is it normal to have a separate namespace for functions intended to be called from the CLI?

Reply:
I change behaviour and make the same way like in your lein plugin realisation